genesis/src/lib.rs

```diff
@@ -47,24 +155,54 @@ impl GenesisNetwork {
         chain_id: &ChainId,
         genesis_key_pair: &KeyPair,
     ) -> GenesisNetwork {
-        // The first instruction should be Executor upgrade.
-        // This makes it possible to grant permissions to users in genesis.
-        let transactions_iter = std::iter::once(GenesisTransactionBuilder {
-            isi: vec![Upgrade::new(raw_block.executor).into()],
-        })
-        .chain(raw_block.transactions);
+        let unified_transaction =
+            GenesisTransaction::new_unified(raw_block, chain_id, genesis_key_pair);
 
-        let transactions = transactions_iter
-            .map(|raw_transaction| raw_transaction.sign(chain_id.clone(), genesis_key_pair))
-            .map(GenesisTransaction)
-            .collect();
+        GenesisNetwork {
+            transaction: unified_transaction,
+        }
+    }
+
+    /// Construct `GenesisSignature` from config
+    /// # Errors
+    /// Fails if the
+    pub fn new_genesis_signature(
+        raw_block: RawGenesisBlock,
+        chain_id: &ChainId,
+        genesis_key_pair: &KeyPair,
+    ) -> GenesisSignature {
```

Comment:
- It is not related to `impl GenesisNetwork`
- `# Errors` is irrelevant

Consider moving it to `RawGenesisBlock`:

```rust
impl RawGenesisBlock {
  pub fn sign(&self, chain_id: &ChainId, key_pair: &KeyPair) -> GenesisSignature;
}
```